### DISCUSSION TITLE
load balance strategy with weight and ip hash combined or weight cookie sticky combined

### DISCUSSION THREAD

**Discussion by cyflhn** [nginx/nginx#260] [General]
I want to know how nginx handle the request when load balance strategy is configured with weight and ip hash or weight and cookie sticky. Can this kind of strategy make sure that the requests from the same ip or same cookie value are routed to the same server?

**Comment by sootie8** [comment]
@cyflhn 
> I want to know how nginx handle the request when load balance strategy is configured with weight and ip hash or weight and cookie sticky.

For IP Hash: weighted hashing.

For Sticky Cookie: If request does not include the (sticky) cookie, upstream peer is selected using previously delcared (in the upstream block) load balancer method before sticky cookie, which is (weighted) round-robin by default. Requests with cookie are routed to server specified by the cookie.

> Can this kind of strategy make sure that the requests from the same ip or same cookie value are routed to the same server?

Yes, both do.
